contracts/allocators/FuseAllocator.sol

```diff
@@ -0,0 +1,539 @@
+}
+
+interface CErc20Interface is CTokenInterface {
+    /*** User Interface ***/
+
+    function mint(uint256 mintAmount) external returns (uint256);
+
+    function redeem(uint256 redeemTokens) external returns (uint256);
+
+    function redeemUnderlying(uint256 redeemAmount) external returns (uint256);
+
+    function borrow(uint256 borrowAmount) external returns (uint256);
+
+    function repayBorrow(uint256 repayAmount) external returns (uint256);
+
+    function repayBorrowBehalf(address borrower, uint256 repayAmount) external returns (uint256);
+
+    function liquidateBorrow(
+        address borrower,
+        uint256 repayAmount,
+        CTokenInterface cTokenCollateral
+    ) external returns (uint256);
+
+    function sweepToken(EIP20NonStandardInterface token) external;
+
+    /*** Admin Functions ***/
+
+    function _addReserves(uint256 addAmount) external returns (uint256);
+}
+
+contract FuseAllocator is OlympusAccessControlled {
+    /* ======== DEPENDENCIES ======== */
+
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    /* ======== STATE VARIABLES ======== */
+    ITreasury public treasury; // Olympus Treasury
+
+    /* ======== CONSTRUCTOR ======== */
+
+    constructor(address _treasury, IOlympusAuthority _authority) OlympusAccessControlled(_authority) {
+        setTreasury(_treasury);
+    }
+
+    /* ======== POLICY FUNCTIONS ======== */
+
+    /**
+     * @notice Deposit the tokens in exchange for the corresponding fTokens,
+     * then enter the designated pool with those fTokens.
+     * @dev The amount of fTokens received is the number of tokens divided by the exchange rate
+     * (https://docs.rari.capital/fuse/#exchange-rate)
+     * @param comptrollerAddress address The address of the desired pool's comptroller.
+     * @param tokenAddress address The address of the token to deposit.
+     * @param fTokenAddress address The address of the fToken corresponding to the given token address.
+     * @param amount uin256 The amount of tokens to deposit.
+     */
+    function deposit(
+        address comptrollerAddress,
+        address tokenAddress,
+        address fTokenAddress,
+        uint256 amount
+    ) external onlyGuardian returns (uint256) {
+        ComptrollerInterface comptroller = ComptrollerInterface(comptrollerAddress);
+        IERC20 token = IERC20(tokenAddress);
+        treasury.manage(tokenAddress, amount); // withdraw specified amount of token from treasury
+
+        token.approve(fTokenAddress, amount); // approve the fToken address to transfer 'amount' in tokenAddress
+
+        CErc20Interface fToken = CErc20Interface(fTokenAddress);
+        uint256 mintAllowed = comptroller.mintAllowed(fTokenAddress, address(this), amount);
+        require(
+            mintAllowed == 0,
+            string(abi.encodePacked("Mint not allowed for specified token and amount, error code ", mintAllowed))
+        );
+        require(fToken.mint(amount) == 0, "Failed to mint fTokens");
```

Comment:
Weird that it uses error codes?

Reply:
For anything that returns an error code yeah

Reply:
I removed the return values on my methods. Do you want me to remove these requires as well? Personally, if they're using error codes I think we should keep these checks in but would love a second opinion.

Reply:
I think its fine to use them in the requires, you're checking that it was succesful, 
Is there any other communication of the error?

Reply:
You mean the specific error that the fuse contract is returning?